06_neighborhoods/Transportation/transit_trips_cost_city.R

```diff
@@ -0,0 +1,292 @@
+
+
+# left join with places file to get rid of irrelevant places data
+transit_trips_city_2019 <- left_join(places, transit_trips_city_2019, by=c("state","place"))
+# 29317 obs to 486 obs
+
+# keep only the variables we will need & replace NA qual vars where there is NA metric value
+transit_trips_city_2019 <- transit_trips_city_2019 %>% 
+  mutate(year==2019)%>%
+  select(year, state, place, index_transit_trips, index_transit_trips_quality)%>%
+  dplyr::mutate(index_transit_trips_quality = ifelse(is.na(index_transit_trips), NA, index_transit_trips_quality))
+
+
+###################################################################
+
+# 3. QC Checks
+# create a histogram plot and summary stats for each dataframe
+# check that all values are non-negative & count missing values
+# examine outliers
+
+
+# City-Level Transportation Cost
+##############################################
+ggplot(transit_cost_city_2015, aes(x=index_transportation_cost)) + geom_histogram(binwidth=10) + labs(y="number of places", x="Annual Transit Cost for the Regional Moderate Income Household, 2015")
+ggplot(transit_cost_city_2019, aes(x=index_transportation_cost)) + geom_histogram(binwidth=10) + labs(y="number of places", x="Annual Transit Cost for the Regional Moderate Income Household, 2019")
+
+# look at summary stats
+summary(transit_cost_city_2015$index_transportation_cost)  
+summary(transit_cost_city_2019$index_transportation_cost)
+# no weird outliers
+
+# Use stopifnot to check if all values in "transit_cost_city_XXXX" are non-negative
+stopifnot(min(transit_cost_city_2015$index_transportation_cost, na.rm = TRUE) >= 0)
+# Good to go
+stopifnot(min(transit_cost_city_2019$index_transportation_cost, na.rm = TRUE) >= 0)
+# Good to go
+
+# Find indices of missing values for the "index_transportation_cost" variable
+missing_indices15 <- which(is.na(transit_cost_city_2015$index_transportation_cost))
+missing_indices19 <- which(is.na(transit_cost_city_2019$index_transportation_cost))
+
+# Print observations with missing values
+if (length(missing_indices15) > 0) {
+  cat("Observations with missing values for index_transportation_cost:\n")
+  print(transit_cost_city_2015[missing_indices15, , drop = FALSE])
+} else {
+  cat("No missing values for transportation_cost_county_2015\n")
+}
+# 5 missing values
+
+if (length(missing_indices19) > 0) {
+  cat("Observations with missing values for index_transportation_cost:\n")
+  print(transit_cost_city_2019[missing_indices19, , drop = FALSE])
+} else {
+  cat("No missing values for transportation_cost_county_2019\n")
+}
+# 5 missing values
+
+
+###################################################################
+
+# 5. Export final files
+
+
+# Combine the two years into one overall files for both variables
+transit_cost_city <- rbind(transit_cost_city_2015, transit_cost_city_2019)
+transit_trips_city <- rbind(transit_trips_city_2015, transit_trips_city_2019)
```

Comment:
Yes, that's right -- because the purpose of creating the ranking is to contextualize it nationally, I felt it made more sense to rank with all the places, then limit to the ones we highlight. I can run this thinking past Claudia if it's worrisome

Reply:
Spoke with Claudia about this -- we feel comfortable with this decision, but we will consult with the team to beef up language explaining this unique situation. Thanks for flagging